crates/bevy_app/src/app.rs

```diff
@@ -288,6 +288,14 @@ impl App {
             panic!("App::run() was called from within Plugin::build(), which is not allowed.");
         }
 
+        if app.ready() {
+            // Force plugins to finish their setup and advance by one frame to make sure everything is
+            // setup up. This is important to prevent black frames during the launch animation on iOS,
+            // but it seems to also solve a brief flicker during startup on Linux/Wayland.
+            app.finish();
+            app.update();
```

Comment:
you need to call `app.cleanup()` between those two call, but that can be called only once and it's already called in winit run loop, so you'll have to handle that